calendar/src/main/java/com/toggl/calendar/datepicker/domain/DatePickerSelector.kt

```diff
@@ -1,10 +1,59 @@
 package com.toggl.calendar.datepicker.domain
 
+import com.toggl.architecture.DispatcherProvider
 import com.toggl.architecture.core.Selector
 import com.toggl.calendar.datepicker.ui.DatePickerViewModel
+import com.toggl.calendar.datepicker.ui.VisibleDate
+import com.toggl.calendar.datepicker.ui.Week
+import com.toggl.environment.services.time.TimeService
+import com.toggl.repository.interfaces.SettingsRepository
+import kotlinx.coroutines.withContext
+import java.time.DayOfWeek
+import java.time.format.DateTimeFormatter
+import javax.inject.Inject
 
-class DatePickerSelector : Selector<CalendarDatePickerState, DatePickerViewModel> {
-    override suspend fun select(state: CalendarDatePickerState): DatePickerViewModel {
-        return DatePickerViewModel(emptyList(), emptyList(), 0)
+class DatePickerSelector @Inject constructor(
+    private val settingsRepository: SettingsRepository,
+    private val timeService: TimeService,
+    private val dispatcherProvider: DispatcherProvider
+) : Selector<CalendarDatePickerState, DatePickerViewModel> {
+    private val daysInAWeek = 7
+    private val dateLabelFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("dd")
+
+    override suspend fun select(state: CalendarDatePickerState): DatePickerViewModel = withContext(dispatcherProvider.io) {
+        val userPreferences = settingsRepository.loadUserPreferences()
+        val userFirstDayOfTheWeek = userPreferences.firstDayOfTheWeek.value - 1
+        val weekLabels = (0 until daysInAWeek)
+            .map { (DayOfWeek.of((it + userFirstDayOfTheWeek) % daysInAWeek + 1)) }
+            .toList()
+
+        val visibleDates = state.visibleDates
+        val availableDates = state.availableDates
+        val selectedDate = state.selectedDate
+        val today = timeService.now()
+
+        val allDates = visibleDates.map {
+            VisibleDate(
+                dateLabelFormatter.format(it),
+                availableDates.contains(it),
+                it.dayOfYear == today.dayOfYear,
+                it
+            )
+        }
+        val weeks = if (allDates.isEmpty()) {
+            emptyList()
+        } else sequence {
+            for (i in allDates.indices step daysInAWeek) {
+                yield(Week(allDates.subList(i, i + daysInAWeek)))
+            }
+        }.toList()
```

Comment:
```suggestion
        } else {
            (allDates.indices step daysInAWeek)
                .map { Week(allDates.subList(it, it + daysInAWeek)) }
        }
```

The sequence looks kinda unnecesery to me here.